src/components/Fragment.js

```diff
@@ -17,20 +17,32 @@ class Fragment extends BaseRouterComponent {
         };
     }
 
+    componentWillReceiveProps(newProps) {
+
+        const { redirect } = this.props;
+        const newRedirect = newProps.redirect;
```

Comment:
'case newRedirect will be `undefined`.
We can make like that, for example:
```js
const { redirect as newRedirect }  = newProps;
```
But I'm not sure what it'll be better.